src/pytest_split/plugin.py

```diff
@@ -134,4 +134,37 @@ def _calculate_suite_start_and_end_idx(splits, group, items, stored_durations):
     if not end_idx:
         end_idx = len(items)
 
+    # Modify start_idx and end_idx so they match start/end of an ipynb file
+    if _is_ipy_notebook(item_node_ids[start_idx]):
+        start_idx = _get_boundary_idx_for_ipynb(item_node_ids, start_idx, "start")
+    if _is_ipy_notebook(item_node_ids[end_idx - 1]):  # since Python indexing is [a, b)
+        end_idx = _get_boundary_idx_for_ipynb(item_node_ids, end_idx, "end")
+
     return start_idx, end_idx
+
+
+def _is_ipy_notebook(node_id):
+    path = node_id.split("::")[0]
+    if path.endswith(".ipynb"):
+        return True
+    return False
+
+
+def _get_boundary_idx_for_ipynb(item_node_ids, idx, mode):
+    if mode not in ["start", "end"]:
+        raise Exception(f"Unsupported mode: {mode}")
+    ipynb_node_id = item_node_ids[idx]
+    fname = ipynb_node_id.split("::")[0]
+    # If node_id provided instead of idx
+    # idx = item_node_ids.index(ipynb_node_id)
+    same_notebook = True
+    while same_notebook:
+        if idx in [0, len(item_node_ids)]:
+            break
+        idx_next = idx - 1 if mode == "start" else idx + 1
+        fname_next = item_node_ids[idx_next].split("::")[0]
+        if fname_next == fname:
+            idx = idx_next
+        else:
+            same_notebook = False
+    return idx if mode == "start" else idx + 1  # since Python indexing is [a, b)
```

Comment:
This is a helper function for obtaining the start/end indices of an IPython notebook, given node_id corresponding to a single cell of that file.